### PR TITLE
Resolving fqdn with "ipfs resolve" and "ipfs name resolve"

### LINKED ISSUE #1918: Resolving DNSLink Paths: /ipns/<fqdn>
> Part of an effort to run embedded js-ipfs in Brave :lion: https://github.com/ipfs-shipyard/ipfs-companion/issues/716

- **Version**: js-ipfs v0.34
- **Platform**: browser
- **Subsystem**: `ipfs.resolve` and `ipfs.name.resolve`

#### Description:

js-ipfs v0.34 supports `/ipns/<peerid>`, but trying to resolve `/ipns/<fqdn>` produces errors (the error is different for each method).

Calling `ipfs.dns('docs.ipfs.io').then(console.log)` resolves DNSLink correctly, so I assume all pieces are in place and this is just a bug/missing some wiring – is this correct?

Is this on someones TODO or is it up for grabs?

#### Steps to reproduce the error:

- `ipfs.resolve('/ipns/docs.ipfs.io/assets/logo.svg', {recursive: true})`
  - go-ipfs returns `/ipfs/QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU`
  - js-ipfs throws `Error: resolve non-IPFS names is not implemented` ([from  this line](https://github.com/ipfs/js-ipfs/blob/v0.34.4/src/core/components/resolve.js#L22-L25))

- `ipfs.name.resolve('/ipns/docs.ipfs.io/assets/logo.svg', {recursive: true})`
  - go-ipfs returns `/ipfs/QmYz85ubdabSrraMnodhMMosiHv9WURJASEC5D6ojTgm4X/assets/logo.svg`
  - js-ipfs throws `Error: Non-base58 character` ([from  this line](https://github.com/ipfs/js-ipfs/blob/v0.34.4/src/core/ipns/resolver.js#L95))

### PR DESCRIPTION
**Pull request by niinpatel**
resolves #1918 

* [x] depends on #1935

### CODE REVIEW

File: test/core/dns.spec.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  after((done) => {
+    if (ipfsd) {
+      ipfsd.stop(done)
+    } else {
+      done()
+    }
+  })
+
+  // skipping because there is an error in https://ipfs.io/api/v0/dns?arg=ipfs.io
+  // unskip once this is resolved: https://github.com/ipfs/go-ipfs/issues/6086
```
Comment (jchris) [inline test/core/dns.spec.js]: this is resolved in https://github.com/ipfs/go-ipfs/pull/6087

Comment (niinpatel) [reply test/core/dns.spec.js]: Hi @jchris, 

When I go to https://ipfs.io/api/v0/dns?arg=ipfs.io   or https://ipfs.io/api/v0/dns?arg=ipfs.io&recursive=false (IPFS companion turned off) It still gives me an error. 

Are the changes from # 6087 integrated into the ipfs.io api?

Comment (jchris) [reply test/core/dns.spec.js]: I don't know, I just know it was closed and merged. Trying to keep the links current.

Comment (niinpatel) [reply test/core/dns.spec.js]: Okay. Thank you. :)

Comment (alanshaw) [reply test/core/dns.spec.js]: It was fairly recent - I think it'll be released in 0.4.20

**Comment by alanshaw** [comment]
Note this needs to take into account IPNS paths can be `/ipns/domain.name` or `/ipns/QmPeerID`. The latter we should now be able to resolve but out of scope for this PR.

**Comment by alanshaw** [comment]
Superseded by https://github.com/ipfs/js-ipfs/pull/2297

**Pull request closed without merging**